var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -414,6 +414,8 @@ def install(self, spec, prefix):
         b2name = './b2' if spec.satisfies('@1.47:') else './bjam'
 
         b2 = Executable(b2name)
+        if "platform=cray" in spec and spec.satisfies("target=mic_knl"):
+            b2 = build_env_compilers(b2)
```

Comment:
You'll want to wrap the `bootstrap.sh` call above vs. the invocation of `b2`: the goal is to create the `b2` executable using the current architecture (vs. the target arch for the package).